R/corpus.R

```diff
@@ -234,54 +230,6 @@ corpus.data.frame <- function(x, docid_field = "doc_id", text_field = "text",
     return(result)
 }
 
-
-#' @rdname corpus
-#' @param split_context logical; if `TRUE`, split each kwic row into two
-#'   "documents", one for "pre" and one for "post", with this designation saved
-#'   in a new docvar `context` and with the new number of documents
-#'   therefore being twice the number of rows in the kwic.
-#' @param extract_keyword logical; if  `TRUE`, save the keyword matching
-#'   `pattern` as a new docvar `keyword`
-#' @examples
-#' # from a kwic
-#' kw <- kwic(data_char_sampletext, "econom*", separator = "",
-#'            remove_separators = FALSE) # keep original separators
-#' summary(corpus(kw))
-#' summary(corpus(kw, split_context = FALSE))
-#' texts(corpus(kw, split_context = FALSE))
-#'
-#' @export
-corpus.kwic <- function(x, split_context = TRUE, extract_keyword = TRUE, meta = list(), ...) {
```

Comment:
Why delete this method for `corpus()`? A lot of users have found this useful for selecting contexts to create a new corpus. In fact I thought you used it for this purpose!

i.e.
https://stackoverflow.com/questions/37445449/how-do-i-transform-a-kwic-from-quanteda-package-to-a-corpus
https://stackoverflow.com/questions/52883002/selecting-sentences-that-contain-specific-words/52883578#52883578

Reply:
There are two reasons. 
1. Users should use `tokens_select(window)` to extract contexts. 
2. User can still use `corpus.data.frame()` with `as.data.frame(kw)`.

Reply:
Perhaps deprecating it first? The corpus.kwic() has some options that those other methods don't, such as split_context and extract_keyword.

Reply:
split_context and extract_keyword can be done with data.frame manually. Demands for `corpus.kwic` is another case for separating the core functions to quanteda.core from quanteda. In this way, quanteda can keep miscellaneous short-cut functions for beginners.

Reply:
Removing this method is not necessary to this PR and should not be part of the PR.

And I think removing it is a bad idea, for several reasons. 

1. Removing it will break existing code unnecessarily.

2. It's not complicating anything else through its existence - nothing else relies on it.

3. Lots of useful actions can be done "manually" using sequences of lower-level steps, but why remove a convenient function like corpus.kwic() that makes it simple and robust? This is super convenient, especially if combined with a way to create a corpus of sentences selected on specific terms. e.g.

```r
> kw <- kwic(data_char_sampletext, "econom*", separator = "", 
+            remove_separators = FALSE)
> 
> corpus(kw, split_context = FALSE)
Corpus consisting of 5 documents and 1 docvar.
text1.L313 :
" the Irish economy in pursuit "

text1.L390 :
" the domestic economy? As we"

text1.L516 :
" the domestic economy show the "

text1.L941 :
" dislocates the economy. Otherwise those"

text1.L976 :
" the domestic economy, stimulating demand"
```

Reply:
Why don't you separate low-level functions to quanteda.core? I have no time or motivation to maintain high-level functions that I do not use (or recommend others to use). If you don't like the name (quanteda.core), I can come up with something very different.

Reply:
Maybe we should name it quanteda.functionskoheidislikes 😄

Reply:
Then I will abandon quanteda and make a new package.

Reply:
My preference would be to make corpus.kwic internal and keep it working with a deprecation warning. It's not doing much harm and the short typical window means converting tokens to a text is not too damaging to performance, despite the philosophical offence.